src/main/java/Parser.java

```diff
@@ -0,0 +1,231 @@
+import java.util.List;
+import java.util.ArrayList;
+
+/*
+ * Deal with making sense of users' commands.
+ */
+public class Parser {
+    private String command;
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+    private boolean isFinished;
+
+    public Parser(TaskList tasklist, Ui ui, Storage storage) {
+        this.command = "";
+        this.taskList = tasklist;
+        this.ui = ui;
+        this.storage = storage;
+        this.isFinished = false;
+    }
+
+    /*
+     * Add a new command to be dealt with.
+     *
+     * @param command New command.
+     */
+    public void insertCommand(String command) {
+        this.command = command;
+    }
+
+    /*
+     * Check if Parser is done dealing with all commands.
+     *
+     * @return Done status.
+     */
+    public boolean isFinished() {
+        return this.isFinished;
+    }
+
+    /*
+     * Deal with the current command.
+     * Pick the appropriate response.
+     * Pass the response to the Ui to handle.
+     * In the case of a bad command, a relevant exception is thrown.
+     * The exception is caught and the relevant message is printed.
+     */
+    public void process() {
+        try {
+            if (command.equals("list")) {
+                ui.replace("Here are the tasks in your list:");
+                for (int i = 0; i < taskList.numOfTasks(); i++) {
+                    Integer taskNum = i + 1;
+                    ui.replace(taskNum.toString() + "." + taskList.getTask(i).toString());
+                }
+            } else if (command.contains("todo")) {
+                String[] line = command.split(" ");
+                int length = line.length;
+                if (length == 1) {
+                    throw new IncompleteException("The description for Todo cannot be empty.");
+                }
+                ui.replace("Got it. I've added this task:");
+                String description = "";
+                for (int i = 1; i < length; i++) {
+                    if (i + 1 == length) {
+                        description += line[i];
+                    } else {
+                        description += line[i] + " ";
+                    }
+                }
```

Comment:
Can the logic used extract out the description be simplified to make it more understandable? Perhaps instead of splitting on spaces and combing all the words after todo, you can simply remove the word 'todo' from the String? This will make require only a single line of code and is much more compact.

Something like:

```
String description  = line.replace("todo", "").strip()  //.strip() is to remove trailing spaces
```

Reply:
Ah yes, that will make it a lot easier, thanks!